src/unleash.ts

```diff
@@ -332,8 +332,15 @@ export class Unleash extends EventEmitter {
     return this.repository.getToggle(toggleName);
   }
 
-  getFeatureToggleDefinitions(): FeatureInterface[] {
-    return this.repository.getToggles();
+  getFeatureToggleDefinitions(
+    withFullSegments = false
```

Comment:
This is quite clever from a perspective of not having to muck with the version number but I'm tempted to bite the bullet and go with this signature instead:

`getFeatureToggleDefinitions(): EnhancedFeatureInterface[]`

It would definitely mean a major version but it'd make the method a lot easier to use. WDYT?

Reply:
The reason of doing it this way is that segment data can be quite massive (seeing how many userIds some customers have as values.
This is meant to solve a use case but maybe overloading the data should be optional. wdyt?

Reply:
Actually hadn't thought of that. Yeah fair enough, that's potentially a very, very expensive nested loop so that would be a surprise to anyone who didn't care about this stuff already. Fair enough!